src/libcore/macros/mod.rs

```diff
@@ -711,6 +711,47 @@ macro_rules! todo {
     ($($arg:tt)+) => (panic!("not yet implemented: {}", $crate::format_args!($($arg)+)));
 }
 
+/// Extracts the successful type of a `Poll<T>`.
+///
+/// This macro bakes in propagation of `Pending` signals by returning early.
+///
+/// # Examples
+///
+/// ```
+/// #![feature(ready_macro)]
+///
+/// use core::task::{Context, Poll};
+/// use core::future::Future;
+/// use core::pin::Pin;
+///
+/// async fn get_num() -> usize {
+///     42
+/// }
+///
+/// pub fn do_poll(cx: &mut Context<'_>) -> Poll<()> {
+///     let mut f = get_num();
+///     let f = unsafe { Pin::new_unchecked(&mut f) };
+///
+///     let num = ready!(f.poll(cx));
+///     # drop(num);
+///     // ... use num
+///
+///     Poll::Ready(())
+/// }
+/// ```
+#[macro_export]
+#[unstable(feature = "ready_macro", issue = "70922")]
+macro_rules! ready {
+    ($e:expr $(,)?) => {
```

Comment:
Is there precedent for an optional trailing comma in macros that don't take comma-separated inputs?

Reply:
Yes, `std`/`core`'s macros support it: #48056
https://github.com/rust-lang/rust/blob/a00bd29e01c4042143d01681e45723a2052412b0/src/libcore/macros/mod.rs#L759-L762

Reply:
Heh that could use `$(,)?`.